ignite/metrics/gan/fid.py

```diff
@@ -0,0 +1,177 @@
+from typing import Callable, Sequence, Union
+
+import numpy as np
+import torch
+
+from ignite.exceptions import NotComputableError
+from ignite.metrics.metric import Metric, reinit__is_reduced, sync_all_reduce
+
+__all__ = ["FID", "InceptionExtractor"]
+
+
+def fid_score(
+    mu1: torch.Tensor, mu2: torch.Tensor, sigma1: torch.Tensor, sigma2: torch.Tensor, eps: float = 1e-6
+) -> float:
+
+    diff = mu1 - mu2
+    import scipy
```

Comment:
This is very unfriendly way to do things. This method is called after 1 epoch done and user get ImportError after some computational time. We should avoid this way of coding things!

Reply:
This just means that we have updated the mean and covariance based on all the training example given by user so far. Even if compute gives error as soon as user imports scipy they can rerun it and no computation done previously is lost.

Reply:
In the case where fid_score function is independently invoked, there is no extra computation so all cases are covered.

Reply:
Please use exception and a user friendly message whether `scipy` is missing.

https://github.com/pytorch/ignite/blob/d16d15efbbbfc476702e91f3ab2bc757b839be26/ignite/contrib/metrics/cohen_kappa.py#L48